src/analyzer/git-caller.server.ts

```diff
@@ -83,7 +84,7 @@ export class GitCaller {
   }
 
   static getCachePath(repo: string, branch: string) {
-    return resolve(__dirname, "..", ".temp", repo, `${branch}.json`)
+    return resolve(os.tmpdir(), "git-truck", repo, `${branch}.json`)
```

Comment:
nice.

Reply:
Can you verify this solves the Docker issue? Then we can also add a working dockerfile.

Reply:
see my comment in the issue.

Reply:
@mircealungu 
I have addressed the issue with unable to open from a headless environment and published a new experimental release `0.0.0-b0a673c`. If you got the time to test it, feel free. I will also try to see if I am able to run it in docker.

Reply:
hmm. 
this is what i get:

```
Git Truck version 0.0.0-b0a673c [!] Update available: 1.2.1

To update, run:

npx git-truck@latest

Or to install globally:

npm install -g git-truck@latest



✔ Git Truck started [745ms]
✔ Done checking for git repo [0ms]
✔ Succesfully opened Git Truck in your browser [11ms]
Error: spawn xdg-open ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:82:21)

```

Reply:
i'm running the following Dockerfile on a freshly installed DigitalOcean droplet

```
#
FROM node:18

# Duplicate deb line as deb-src
RUN cat /etc/apt/sources.list | sed "s/^deb\ /deb-src /" >> /etc/apt/sources.list

# Install common build utilities
RUN apt update
RUN DEBIAN_FRONTEND=noninteractive apt install -yy eatmydata
RUN DEBIAN_FRONTEND=noninteractive eatmydata \
    apt install -y --no-install-recommends \
        fish \
        git \
        zile

# We can't set the USER/UID as the source docker image already maps
# stuff to something.

WORKDIR /usr/app
RUN npm install git-truck@0.0.0-b0a673c 

```

followed by running:

    docker run --rm -it -u `id -u` -v `pwd`:`pwd` -w `pwd` -p 3000:3000 gt /usr/app/node_modules/.bin/git-truck

Reply:
Hmm strange. It's hard for me to reproduce, I couldn't get the docker image to run.

Reply:
I also can't run it on mac os with an M1 chip... this is why I created a new droplet and tried it there. Based on the `Succesfully opened Git Truck in your browser` message it seems that the code thinks that it could actually open the browser after all?

Reply:
Oh wow, I didn't notice that. That's strange, that it then still throws an exception.

Reply:
I managed to reproduce this by spawning a child process that exits with a nonzero exit code
```
✔ Git Truck started [3.88s]
✔ Done checking for git repo [152ms]
✔ Succesfully opened Git Truck in your browser [4ms]                                                                              
Error: spawn exit 1 ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:477:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

Reply:
I have found others facing similar issues. Can you try adding
```
apt install xdg-utils --fix-missing
```
to the dockerfile?

https://stackoverflow.com/questions/73942710/vitest-ui-causing-error-spawn-xdg-open-enoent

Reply:
Oh nvm, it [should be included already](https://github.com/sindresorhus/open#why) with the package...

Reply:
Maybe you're right, maybe we should just add a `--headless` flag afterall

Reply:
I've added a `--headless` argument now

@mircealungu Can you try with this experimental release: `git-truck@0.0.0-de7c4e6`

Reply:
this works @joglr 

it's temporarily deployed at:
http://46.101.192.120:3000/git-truck/main

from a docker container :)

Reply:
Cool! How do we proceed then, do we add a dockerfile to the repo? Or do we need to setup something to build container images?

Reply:
I think Dockerfile should suffice? It wasn't such a frequent request that we should try to automatically build an image...

Reply:
I agree. Can you add the working dockerfile to this branch?